libcxx/include/__filesystem/directory_entry.h

```diff
@@ -201,7 +201,11 @@ class directory_entry {
     _IterNonSymlink,
     _RefreshSymlink,
     _RefreshSymlinkUnresolved,
-    _RefreshNonSymlink
+    _RefreshNonSymlink,
+#  if _LIBCPP_AVAILABILITY_HAS_FILESYSTEM_FULLY_POPULATED_CACHED_ENTRY
```

Comment:
I have just been thinking about should we really use define `_LIBCPP_AVAILABILITY_HAS_FILESYSTEM_FULLY_POPULATED_CACHED_ENTRY`
Because If I add `_RefreshNonSymlink` and ` _RefreshNonSymlink`
   in the end of enum it will mean that the new filesystem source code in trying to use the old dylib will stay correct because the old dylib will use old enum values in filling directory_entry and enum order didnt' change.

Reply:
I had thoughts about reusing `_IterSymlink` and `_IterNonSymlink` in my first attempt to add caching and I faced that I would be kinda difficult because in that case I should store information about cached or not properties depending on platform Windows and others. And also based on that knowledge rewrite many functions with enum switch by adding for each case with  `_IterSymlink` and `_IterNonSymlink` additional login which quite similar to  '_IterCachedSymlink' and '_IterCachedNonSymlink'
In that case each plain enum switch will become enum with additional if else.

Reply:
I have prepared patch without using availability define. If it suits for you we manage to release this patch with LLVM 19. Otherwise I can try to rework this patch like in the first option and release it later.